### ISSUE TITLE
Memory error due to large hdf5 file

### ISSUE THREAD

**Issue by mhasnat** [tflearn/tflearn#101]
Hi,

I have encountered memory error, most possible due to large (5M color image, consists of 42GB) h5 input file. The data is loaded successfully by the command:
X = h5f['dataImg'][()]
Y = h5f['label']

However, I simply get memory error when I run the following command:
Y = to_categorical(Y, numClass)

This problem does not occur when the data size is smaller. I would like to know a possible solution for this problem. It seems that this problem is not very specific to TFLearn. However, yet I am asking for help/suggestion.

Thank you.

**Comment by aymericdamien** [comment]
Yes, it is because to_categorical will actually load data into memory (as numpy array).
Either you can convert your hdf5 lab dataset to logits, or as a hack manually add to `utils.py` source code: https://github.com/tflearn/tflearn/blob/master/tflearn/utils.py#L239
```
Y = tflearn.data_utils.to_categorical(Y, num_classes)
```

To improve TFLearn, we should find a way to let users easily convert their integer labels to logits (Maybe in regression layer or elsewhere).

**Comment by mhasnat** [comment]
Thank you for your answer. I think the problem is with variable X which consumes a large memory due to loading 5M color images. Comparatively Y is very small and also the function to_categorical() does not need to access/load X.

I am going to follow your feedback.
